pkg/operator/controller/ingress/status.go

```diff
@@ -925,6 +941,35 @@ func computeLoadBalancerProgressingStatus(ic *operatorv1.IngressController, serv
 	}
 }
 
+// updateIngressControllerAWSSubnetStatus mutates the provided IngressController object to
+// sync its status to the effective subnets on the LoadBalancer-type service.
+func updateIngressControllerAWSSubnetStatus(ic *operatorv1.IngressController, service *corev1.Service) {
+	// Set the subnets status based on the actual service annotation and the load balancer type,
+	// as NLBs and CLBs have separate subnet configuration fields.
+	switch getAWSLoadBalancerTypeInStatus(ic) {
+	case operatorv1.AWSNetworkLoadBalancer:
+		// NetworkLoadBalancerParameters will be initialized by setDefaultPublishingStrategy
```

Comment:
Then they cannot be `nil`, no? Do we need the `if` above?

Reply:
`Type` is `required` but NetworkLoadBalancerParameters is `optional` and of type pointer so i think it is better to check nil for it from my understanding.